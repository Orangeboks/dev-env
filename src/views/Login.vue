<template>
  <div>
    <form>
      <div class="container">
        <input
          type="email"
          v-model="email"
          placeholder="Enter email"
          name="email"
          required
        />
        <input
          type="password"
          v-model="password"
          placeholder="Enter Password"
          name="psw"
          required
        />
        <button type="submit" @click.prevent="login()">Login</button>
      </div>
    </form>
  </div>
</template>

<script>
import * as firebase from "firebase";
// @ is an alias to /src

export default {
  name: "Login",

  data() {
    return {
      group_chat: true,
      receiver_uid: "",
      message: null,
      private_message: "",
      private_messages: [],
      messages: [],
      authUser: [],
      users: [],
      email: "",
      password: ""
    };
  },
  created() {
    if (localStorage.getItem("login_user_email")) {
      this.$router.push("/");
    }
  },
  methods: {
    login() {
      if (this.email && this.password) {
        firebase
          .database()
          .ref()
          .child("users/")
          .once("value", snapshot => {
            var users = snapshot.val();
            for (var user in users) {
              var user_data = users[user];
              if (
                this.email === user_data.email &&
                this.password === user_data.password
              ) {
                localStorage.setItem("login_user_name", user_data.name);
                localStorage.setItem("login_user_email", user_data.email);
                this.$router.push("UploadVideo");
                location.reload();
                break;
              } else {
                alert("Incorrect email or password");
              }
            }
          });
      } else {
        alert("Enter valid email and password");
      }
    },
    fetchUsers() {
      firebase
        .database()
        .ref()
        .child("users/")
        .once("value", snapshot => {
          var users = snapshot.val();
          console.log(users);
          //    var allRegisteredUsers = [];
          //    for(var x in users){
          //      var user =users[x];
          //      allRegisteredUsers.push(user);
          //    }
          //    this.users = allRegisteredUsers;
        });
    }
  }
};
</script>

<style>
body {
  font-family: Arial, Helvetica, sans-serif;
}
form {
  border: 3px solid #f1f1f1;
}

input[type="email"],
input[type="password"] {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  box-sizing: border-box;
}

button {
  background-color: #4caf50;
  color: white;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  cursor: pointer;
  width: 100%;
}

button:hover {
  opacity: 0.8;
}

.container {
  padding: 16px;
}

span.psw {
  float: right;
  padding-top: 16px;
}

/* Change styles for span and cancel button on extra small screens */
@media screen and (max-width: 300px) {
  span.psw {
    display: block;
    float: none;
  }
  .cancelbtn {
    width: 100%;
  }
}
</style>
