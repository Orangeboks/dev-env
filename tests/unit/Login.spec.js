import { shallowMount } from '@vue/test-utils';
import Login from '../../src/views/Login.vue';
import './common/localStorageMock';
jest.mock('firebase', () => {
	const set = jest.fn();

	return {
		database: jest.fn(() => ({
			ref: jest.fn(() => ({
				push: jest.fn(() => ({
					set,
				})),
				 child: jest.fn(() => ({
				 	once: jest.fn((key, cb) => {
				 		const value = {};
				 		value.val = jest.fn(() => [
				 			{
				 				'1': {
				 					email: 'test@test.com',
				 					password: '1234',
				 				},
				 			},
				 		]);
				 		cb(value);
				 	}),
				 })),
			})),
		})),
	};
});

describe('Login.vue', () => {
	const wrapper = shallowMount(Login);
	it('renders li for each item in props.items', () => {
		expect(wrapper.findAll('.container input')).toHaveLength(2);
		expect(wrapper.findAll('.container button')).toHaveLength(1);
	});

	it('matches snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	 it('login method success', () => {
	 	wrapper.setData({
	 		email: 'test@test.com',
	 		password: '1234',
	 	});
	 	wrapper.vm.login();
	 });

	 it('login method failure', () => {
	 	wrapper.setData({
	 		email: null,
	 		password: '1234',
	 	});
	 	wrapper.vm.login();
	 });
});
