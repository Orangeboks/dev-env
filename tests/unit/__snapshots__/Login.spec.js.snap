// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue matches snapshot 1`] = `
<div>
  <form>
    <div class="container"><input type="email" placeholder="Enter email" name="email" required="required"> <input type="password" placeholder="Enter Password" name="psw" required="required"> <button type="submit">Login</button></div>
  </form>
</div>
`;
